### PR TITLE
add employment contract and unit test

### PR DESCRIPTION
**Pull request by rantifamilusi**
# Description

Add Employment for searchApi contract.
[FAMS3-1483] (https://justice.gov.bc.ca/jira/browse/FAMS3-1483)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested? 

Unit test
# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: **FAMS3-1483**

### CODE REVIEW

File: app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employment.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    public class Employment : PersonalInfo
+    {
+        public string IncomeAssistance { get; set; }
+
+        public string IncomeAssistanceStatus { get; set; }
+
+        public Employer Employer { get; set; }
+
+        public string Occupation { get; set; }
+
```
Comment (peggy-gov-bc-ca) [inline app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employment.cs]: missing  employmentConfirmed ?

File: app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employer.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        public string PhoneExtension { get; set; }
+
+        [Description("The Fax number of the employer")]
+        public string FaxNumber { get; set; }
+
+        [Description("The extension to the fax number of the employer")]
+        public string FaxExtension { get; set; }
+
+        [Description("The address of the employer")]
+        public Address Address { get; set; }
```
Comment (peggy-gov-bc-ca) [inline app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employer.cs]: shall we do the same for Phone? Maybe dynamics team can provide more input.

File: app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employment.cs (hunk 1 above)
Comment (rantifamilusi) [reply app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employment.cs]: I think it's funny that you have employment information and ask if employment is confirmed. That field is redundant to me

File: app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employer.cs (hunk 2 above)
Comment (rantifamilusi) [reply app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employer.cs]: I don't understand.. what should we do for phone ?

File: app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employment.cs (hunk 1 above)
Comment (peggy-gov-bc-ca) [reply app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employment.cs]: If it is in ICBC data, we need to transfer it. that's what I think.

Comment (rantifamilusi) [reply app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employment.cs]: We don't want to be tightly coupled with ICBC though. The presence of the data is a confirmation of employment except the fields means something else

Comment (peggy-gov-bc-ca) [reply app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employment.cs]: We are not sure all the employments are confirmed.

Comment (rantifamilusi) [reply app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employment.cs]: Maybe I don't understand, what does that field mean ? that someone is employed or not ?

File: app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employer.cs (hunk 2 above)
Comment (peggy-gov-bc-ca) [reply app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employer.cs]: I mean if we use Address class, for phonenumber and extension, we could also use Phone class

Comment (rantifamilusi) [reply app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employer.cs]: Oh I don't recommend ... phone is a lot more that just number and extension. The address attached to Employer is the same as main address except for type which can be defaulted.

File: app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employment.cs (hunk 1 above)
Comment (peggy-gov-bc-ca) [reply app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employment.cs]: i think icbc got employment info , for example, from applicant. But this employment info is not confirmed through other ways. So, search officer should not be 100% sure this the correct employment info.

Comment (rantifamilusi) [reply app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employment.cs]: That makes sense. Information may not be correct. Would update

However we don't have that information on the UI (current UI). It still seems like if employment information is not real, it won't be available. We should be careful not to be tightly coupled with data providers schema cos we would be creating technical debt.

Comment (rantifamilusi) [reply app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employment.cs]: Updated

File: app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employer.cs (hunk 2 above)
Comment (peggy-gov-bc-ca) [reply app/SearchApi/BcGov.Fams3.SearchApi.Contracts/Person/Employer.cs]: sounds good.

**Pull request merged**